autogpt_platform/frontend/src/components/integrations/credentials-input.tsx

```diff
@@ -148,14 +151,69 @@ export const CredentialsInput: FC<{
     }
   }, [credentials, selectedCredentials, onSelectCredentials]);
 
-  const singleCredential = useMemo(() => {
-    if (!credentials || !("savedCredentials" in credentials)) return null;
+  // Helper function to extract host from URL
+  const getHostFromUrl = (url: string): string | null => {
+    try {
+      const urlObj = new URL(url);
+      return urlObj.hostname;
+    } catch {
+      return null;
+    }
+  };
```

Comment:
These type of helpers can go into `src/lib/utils/url.ts` for example, as it is likely we will need them in the future in other contexts 👍🏽

Reply:
Updated